packages/vscode-extension/src/project/project.ts

```diff
@@ -266,7 +267,13 @@ export class Project
       }
     }
   }
-  //#endregion
+
+  public async reload(type: ReloadAction): Promise<boolean> {
+    this.updateProjectState({ status: "starting" });
+    const success = (await this.deviceSession?.perform(type)) ?? false;
+    this.updateProjectState({ status: "running" });
```

Comment:
is this ok that we always move to running state after reload even if it doesn't succeed?

Reply:
For now, we assume it can't fail. We'll revisit after we adapt the lower API to return booleans on success instead of throwing exceptions. I'll add a comment here.